# Using DeepDive for Gene Variation Inference
Another Natural Language Processing (NLP) exploration in Bioinformatic literatures in 2017 autumn semester. The previous exploration in 2016 autumn could refer [here](https://github.com/az7jh2/NLP-Exploration-in-Bioinformatics).

This is a Neural Information Processing Systems (**NIPS**) [2017 Competition Track](https://nips.cc/Conferences/2017/CompetitionTrack) named "**Classifying Clinically Actionable Genetic Mutations**", which was hosted by [Kaggle](https://www.kaggle.com/c/msk-redefining-cancer-treatment).

To predict the effect of Genetic Variants to enable Personalized Medicine, we need to determine oncogenicity (4 classes) and mutation effect (9 classes) of the genes based on abstracts of biomedical literatures. It can be defined as a **knowledge discovery** task, i.e. gene-variation-effect relationship. I tried [DeepDive](http://deepdive.stanford.edu), which has been stopped updating from 2017.

The detailed procedures could refer ppt file.

In this work, other skills were also needed:
* A special program language *DDlog* presented by author
* Other programming skills: *bash*, *Python*, *SQL*, *HTML*
* A database tool *PostgreSQL*
